linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -164,6 +165,37 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
     }
   }
 
+  private def checkDuplicateMembers(classDef: LinkedClass): Unit = {
+    // Fields
+    val declaredFields = mutable.Set.empty[(Boolean, FieldName)]
+    for {
+      field @ FieldDef(flags, FieldIdent(name, _), _) <- classDef.fields
+    } {
+      implicit val ctx = ErrorContext(field)
+
+      if (!declaredFields.add((flags.namespace.isStatic, name))) {
+        reportError(
+            i"Duplicate definition of field '$name' in class " +
+            i"'${classDef.encodedName}'")
+      }
+    }
```

Comment:
Have you considered simply using a group by?

```scala
for {
  (id, dupes) <- classDef.fields.groupBy(f => (f.flags.namespace, f.name.name))
  if dupes.size > 1
} {
  reportError(...)
}
```

Reply:
Not sure it's better, but I've applied your suggestion.

Reply:
Indeed, that's better. Updated.